cmd/mt-whisper-importer-writer/main.go

```diff
@@ -0,0 +1,283 @@
+			}
+		}
+
+		archiveTTL := a.ArchiveInfo.SecondsPerPoint * a.ArchiveInfo.Points
+		tableTTL := s.selectTableByTTL(archiveTTL)
+		tableName, ok := mdata.GetTableByTTL(s.TTLTables, tableTTL)
+		if !ok {
+			throwError(fmt.Sprintf("Failed to find table with ttl %d in %+v", tableTTL, s.TTLTables))
+		}
+
+		if !avg || archiveIdx == 0 || !*fakeAvgAggregates {
+			log(fmt.Sprintf(
+				"inserting %d chunks of archive %d with ttl %d into table %s with ttl %d and key %s",
+				len(a.Chunks), archiveIdx, archiveTTL, tableName, tableTTL, a.MetricData.Id,
+			))
+			s.insertChunk(tableName, a.MetricData.Id, tableTTL, a.Chunks)
+		} else {
+			// averaged archives are a special case because mt doesn't store them as such.
+			// mt reconstructs the averages on the fly from the sum and cnt series, so we need
+			// to generate these two series out of raw averaged data by multiplying each point
+			// with the aggregation span and storing the result as sum, cnt is the aggregation span.
+			type sumCntChunks struct {
+				sum *chunk.Chunk
+				cnt *chunk.Chunk
+			}
+
+			// seconds per point is assumed to be the aggregation span
+			aggSpan := a.ArchiveInfo.SecondsPerPoint
+
+			sumArchive := make(archive.ArchiveOfChunks)
+			cntArchive := make(archive.ArchiveOfChunks)
+			for T0, c := range a.Chunks {
+				sum := chunk.New(T0)
+				cnt := chunk.New(T0)
+
+				it, err := tsz.NewIterator(c.Bytes)
+				if err != nil {
+					throwError(fmt.Sprintf("failed to get iterator from chunk: %s", err))
+				}
+				for it.Next() {
+					ts, val := it.Values()
+					cnt.Push(ts, float64(aggSpan))
+					sum.Push(ts, val*float64(aggSpan))
+				}
+
+				cnt.Finish()
+				sum.Finish()
+
+				sumArchive[T0] = archive.MetricChunk{ChunkSpan: aggSpan, Bytes: sum.Bytes()}
+				cntArchive[T0] = archive.MetricChunk{ChunkSpan: aggSpan, Bytes: cnt.Bytes()}
+			}
+
+			cntId := api.AggMetricKey(base_id, "cnt", aggSpan)
+			sumId := api.AggMetricKey(base_id, "sum", aggSpan)
+
+			log(fmt.Sprintf(
+				"inserting 2 archives of %d chunks per archive with ttl %d into table %s with ttl %d and keys %s/%s",
+				len(a.Chunks), archiveTTL, tableName, tableTTL, cntId, sumId,
+			))
+
+			s.insertChunk(tableName, cntId, tableTTL, cntArchive)
+			s.insertChunk(tableName, sumId, tableTTL, sumArchive)
+
+		}
+	}
+}
+
+func (s *Server) insertChunk(table, id string, ttl uint32, chunks archive.ArchiveOfChunks) {
+	query := fmt.Sprintf("INSERT INTO %s (key, ts, data) values (?,?,?) USING TTL %d", table, ttl)
+	for t0, chunk := range chunks {
+		row_key := fmt.Sprintf("%s_%d", id, t0/mdata.Month_sec)
+		err := s.Session.Query(query, row_key, t0, mdata.PrepareChunkData(chunk.ChunkSpan, chunk.Bytes)).Exec()
+		if err != nil {
+			throwError(fmt.Sprintf("Error in query: %s", err))
+		}
+	}
+}
+
+func (s *Server) selectTableByTTL(ttl uint32) uint32 {
+	selectedTTL := uint32(math.MaxUint32)
+
+	// find the table with the smallest TTL that is at least equal to archiveTTL
```

Comment:
Because if we do that then metrictank won't be able to find it. The Assumption is that Metrictank is using a defined `agg-settings`, we want to import the data into that schema while making as much sense as possible. It would be possible to create the right tables and insert the data there, but then MT's `agg-settings` might have to be changed to accommodate the data.